treecorr/catalog.py

```diff
@@ -1499,12 +1513,19 @@ def set_patch(data, patch_col):
             # We should be able to always use s = slice(self.start, self.end, self.every_nth)
             if self.start == 0 and self.end is None and self.every_nth == 1:
                 s = slice(None)
+            # fancy indexing in h5py is incredibly slow, so we explicitly
+            # check if we can slice or not.  This checks for the fitsio version in
+            # the fits case
+            elif infile.can_slice:
+                s = slice(self.start, self.end, self.every_nth)
             else:
```

Comment:
This is not covered now.  If you want to try, I think it needs a `mock` test that sets the fitsio version to 1.0.6 (or mocking FitsReader.can_slice to return False).  But if you want to punt on that, you can just put `#pragma: no cover` here with a comment that this is only needed for fitsio <= 1.0.6.